ekotrace-udp-collector/src/chunked.rs

```diff
@@ -0,0 +1,199 @@
+use chrono::{DateTime, Utc};
+use std::collections::{BTreeMap, HashMap};
+use util::alloc_log_report::LogReport;
+
+pub struct ChunkHandlingConfig {
+    pub max_total_open_time: chrono::Duration,
+    pub max_time_since_last_chunk: chrono::Duration,
+}
+
+impl Default for ChunkHandlingConfig {
+    fn default() -> Self {
+        ChunkHandlingConfig {
+            max_total_open_time: chrono::Duration::seconds(600),
+            max_time_since_last_chunk: chrono::Duration::seconds(120),
+        }
+    }
+}
+
+pub struct ChunkHandler {
+    config: ChunkHandlingConfig,
+
+    #[allow(clippy::type_complexity)]
```

Comment:
Man, I'm inclined to agree with clippy here. At least a comment as to the meaning of these different map keys would be welcome.

Reply:
Righto, will clarify.